src/conn/conn_tiered.c

```diff
@@ -216,6 +216,12 @@ __tier_do_operation(WT_SESSION_IMPL *session, WT_TIERED *tiered, uint32_t id, co
         if (ret == ENOENT)
             ret = 0;
         else if (ret == 0) {
+            /* Cache the flushed content into chunkcache. */
```

Comment:
I think the flag check should be done in `__wt_chunkcache_ingest` - I don't like that other bits of code are reaching into the cache and checking config flags. (I realise other code does it with `WT_CHUNKCACHE_CONFIGURED`, but that doesn't mean we should do it too.)

Is the concern that `__wt_tiered_name` shouldn't be called if we won't use the result?

Reply:
I agree with the reasoning @korteland. However, the tricky bit here is that `__wt_tiered_name` function call becomes unnecessary if chunkcache is not configured (if we are checking the flags in __wt_chunkcache_ingest). To get around that we could call `__wt_tiered_name` in the `__wt_chunkcache_ingest` function, but we need access to the tiered dhandle (&tiered->iface) in that case. 

I propose that we either check these flags in both places or pass the tiered dhandle in __wt_chunkcache_ingest. I prefer the first option. Let me know what you think.

Reply:
That's fair enough. I did a little more reading, and it appears that `__wt_tiered_name` will allocate memory. To me, that changes things in favour of checking the enabled flag. I wouldn't check the "enable ingestion" flag - it'll be on most of the time, and it feels more reasonable to have the "chunk cache enabled" flag be part of the chunk cache API than the "enable ingestion" flag. (Just to clarify though, I think `__wt_chunkcache_ingest` should still check both flags.)

Reply:
FYI, with the `WT_TIERED_NAME_SKIP_PREFIX` flag passed in `__wt_tiered_name`, we wont be allocating memory.

Reply:
Got it - thanks for the correction.